### PR TITLE
XYChart: Add all dataset columns in tooltip

### LINKED ISSUE #64450: XY Chart: Add all columns to tooltip in legend
**What happened**:
Upon clicking a point, not all series are included in the tool-tip - in particular, the `Current Year Population` field is working great to generate the size of the point/circle, but it's not showing up in the tool-tip.

Similarly, the `Country Name` isn't showing up, which would be important to identify which circle I'm clicking

**What you expected to happen**:
See above

**How to reproduce it (as minimally and precisely as possible)**:
https://play.grafana.org/d/_KUTfkjnk/gapminder-ish-hans-rosling?orgId=1&viewPanel=20

**Anything else we need to know?**:
image

**Environment**:
- Grafana version: 9.4.3
- Data source type & version: BigQuery
- OS Grafana is installed on: Grafana CLoud
- User OS & Browser:
- Grafana plugins:
- Others:

### LINKED ISSUE #65042: XY Chart: problems with tooltip
Dear grafana community,

grafana is no longer able to display all values of the query in the tooltip compared to v9.1.6. Previously, additional queried columns were displayed in the tooltip. With 9.4.3, only the x and y values are displayed, but no additional query values.

See [this screenshot](https://i.ibb.co/LJKz03F/comparison-grafana.png), performed with the same data but the different Grafana versions.

@leeoniya mentioned this in already in #57221 (issue was closed by mistake)

Thank you

### PR DESCRIPTION
**Pull request by mdvictor**
Screenshot 2023-03-20 at 10 59 58

**What is this feature?**

Fixes a bug where facets in tooltip would not show values if the field name was overriden.
Applies correct name (e.g overriden name) of the y-value in the tooltip.
Shows extra fields in the tooltip

**Why do we need this feature?**

Enchances the xy chart tooltip and fixes bugs.

**Who is this feature for?**

Everyone

**Which issue(s) does this PR fix?**:

Fixes #64450
Fixes #65042

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/plugins/panel/xychart/TooltipView.tsx [hunk 1]
```diff
@@ -101,8 +106,8 @@ export const TooltipView = ({
         </tr>
         <tbody>
           <tr>
-            <th>{xField.name}</th>
-            <td>{fmt(xField, xField.values.get(rowIndex))}</td>
+            <th>{getFieldDisplayName(xField, frame)}</th>
+            <td>{fmt(frame.fields[0], xField.values.get(rowIndex))}</td>
```
Comment (leeoniya) [inline public/app/plugins/panel/xychart/TooltipView.tsx:110]: not sure i understand this change. why `frame.fields[0]` instead of `xField`? :thinking:

Comment (nmarrs) [reply public/app/plugins/panel/xychart/TooltipView.tsx:110]: Reverting these changes back to original results in following

Screenshot 2023-03-27 at 9 26 43 AM

With current changes it seems we lose % context for the x axis

Screenshot 2023-03-27 at 9 27 05 AM


Given this I'm not sure on why we need this change as well

**Review by nmarrs — APPROVED** [review]
Works well locally, I think we should revisit the changes Leon called out as I'm not sure why they are necessary / if they are necessary

File: public/app/plugins/panel/xychart/TooltipView.tsx (hunk 1 above)
Comment (mdvictor) [reply public/app/plugins/panel/xychart/TooltipView.tsx:110]: Yes, that looks like a mistake. Clearly it should be `xField`. I will modify and re-check that everything is ok

**Comment by mdvictor** [comment]
@leeoniya @nmarrs fixed that small bug. should be good now

**Review by leeoniya — APPROVED** [review]

**Pull request merged**